core/src/main/java/org/apache/iceberg/io/RollingEqualityDeleteWriter.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.io;
+
+import org.apache.iceberg.FileFormat;
+import org.apache.iceberg.PartitionSpec;
+import org.apache.iceberg.StructLike;
+import org.apache.iceberg.deletes.EqualityDeleteWriter;
+import org.apache.iceberg.encryption.EncryptedOutputFile;
+
+/**
+ * A rolling equality delete writer that splits incoming deletes into multiple files within one spec/partition.
+ */
+public class RollingEqualityDeleteWriter<T> extends RollingDeleteWriter<T, EqualityDeleteWriter<T>> {
```

Comment:
Is there necessary to share the abstracted `RollingDeleteWriter` between `RollingEqualityDeleteWriter` & `RollingPositionDeleteWriter` ?  The key common thing is the `referencedDataFiles`  I think,  but the `RollingEqualityDeleteWriter` usually don't produce any referenced data files, right ?  So I think we may could just make the RollingEqualityDeleteWriter extend the RollingFileWriter.

Reply:
The idea was to avoid implementing `addResult` and `aggregatedResult` in both but now that I look at it, it is probably not worth a separate class. I'll remove it.